test/jsonapi/plugs/content_type_negotiation_test.exs

```diff
@@ -4,35 +4,101 @@ defmodule JSONAPI.ContentTypeNegotiationTest do
 
   alias JSONAPI.ContentTypeNegotiation
 
-  test "halts and returns an error if content-type header is incorrect" do
+  test "passes request through" do
+    conn =
+      :post
+      |> conn("/example", "")
+      |> Plug.Conn.put_req_header("content-type", "application/vnd.api+json")
+      |> Plug.Conn.put_req_header("accept", "application/vnd.api+json")
+      |> ContentTypeNegotiation.call([])
+
+    refute conn.halted
+  end
+
+  test "halts and returns an error if no content-type or accept header" do
     conn =
       :post
       |> conn("/example", "")
       |> ContentTypeNegotiation.call([])
 
+    refute conn.halted
+  end
+
+  test "passes request through if only content-type header" do
+    conn =
+      :post
+      |> conn("/example", "")
+      |> Plug.Conn.put_req_header("content-type", "application/vnd.api+json")
+      |> ContentTypeNegotiation.call([])
+
+    refute conn.halted
+  end
+
+  test "passes request through if only accept header" do
+    conn =
+      :post
+      |> conn("/example", "")
+      |> Plug.Conn.put_req_header("accept", "application/vnd.api+json")
+      |> ContentTypeNegotiation.call([])
+
+    refute conn.halted
+  end
+
+  test "passes request through if multiple accept header" do
+    conn =
+      :post
+      |> conn("/example", "")
+      |> Plug.Conn.put_req_header("accept", "application/vnd.api+json, application/vnd.api+json; version=1.0")
+      |> ContentTypeNegotiation.call([])
+
+    refute conn.halted
+  end
+
+  test "halts and returns an error if content-type header contains other media type" do
+    conn =
+      :post
+      |> conn("/example", "")
+      |> Plug.Conn.put_req_header("content-type", "text/html")
+      |> ContentTypeNegotiation.call([])
+
+    assert conn.halted
+    assert 415 == conn.status
+  end
+
+  test "halts and returns an error if content-type header contains other media type params" do
+    conn =
+      :post
+      |> conn("/example", "")
+      |> Plug.Conn.put_req_header("content-type", "application/vnd.api+json; version=1.0")
+      |> Plug.Conn.put_req_header("accept", "application/vnd.api+json")
+      |> ContentTypeNegotiation.call([])
+
```

Comment:
I think I covered all the test cases here.  Lmk if you see something more! :thought_balloon: